Bug fix: The restoration activity log index broke if there were any log entrie…

LizPrescott opened the pull request:
…s that we not associated with a nursery table. An alternative to this quick fix is to enforce the presence of a nursery table id (it is marked optional: true in the restoration activity log entry model). Would also need to figure out why the nursery table is not being assigned in the create new log form.

<!--Read comments, before commiting pull request read checklist again



### Description
<!-- Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context.
Guide questions:
  - What motivated this change (if not already described in an issue)?
  - What alternative solutions did you consider?
  - What are the tradeoffs for your solution?
   
List any dependencies that are required for this change. (gems, js libraries, etc.)

Include anything else we should know about. -->

### Type of change

<!-- Please delete options that are not relevant. -->

* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Do we need to do anything else to verify your changes? 
If so, provide instructions (including any relevant configuration) so that 
we can reproduce? -->
Tested manually in the browser and added specs to confirm that table_name produced the desired output in both major cases. 

### Screenshots
<!--Optional. Delete if not relevant. 
Include screenshots (before / after) for style changes, highlight 
edited element.-->

Commit:
The restoration activity log index broke if there were any log entries that we not associated with a nursery table. An alternative to this quick fix is to enforce the presence of a nursery table id (it is marked optional: true in the restoration activity log entry model). Would also need to figure out why the nursery table is not being assigned in the create new log form.

cflipse commented on rails/app/views/restoration_activity_log_entries/index.html.erb:
```diff
@@ -5,7 +5,7 @@
       <div class="card text-center">
         <div class="card-body restoration-index">
-          <h5 class="card-title"><%= entry.nursery_table.name %></h5>
+          <h5 class="card-title"><%= entry.nursery_table&.name %></h5>
```
Can we move this (`&.`) logic into the model, say as a `table_name` (maybe use a delegator with allow_nil) so that we don't have to manage to replicate this logic in every view?

LizPrescott replied:
yep. That sounds more readable too. Working on this now.

cflipse commented:
The biggest risk of doing a validation on the presence of the table_id is that ... it can still change and get lost  (think, a table row gets deleted from the db) or otherwise.

A `table_name` method in the log might be more robust, and allows for a fallback of "not found" / unknown if it can't make the link, instead of just propagating a nil

Commit:
table_name method on Entries model

Instead of relying on the presence of a parent nursery_table to populate views related to entries, views should use this method.
